### PR TITLE
Add cargo-audit test to travis.yml

### LINKED ISSUE #5: Implement cargo-audit testing for new pull requests
We'd like to enforce [`cargo-audit`](https://blog.rust-lang.org/inside-rust/2019/10/03/Keeping-secure-with-cargo-audit-0.9.html) testing on all new pull requests via Travis. This should be an augmentation to the existing Travis configuration.

### PR DESCRIPTION
**Pull request by mbestavros**
Resolves #5.

### CODE REVIEW

**Comment by mbestavros** [comment]
@steveeJ I've fixed the config to prevent Rust 1.31 from attempting to run `cargo-audit` (which was not supported until 1.36, and was therefore causing failures).

Ready for review.

File: .travis.yml [hunk 1]
```diff
@@ -24,5 +24,11 @@ matrix:
   fast_finish: true
 install:
   - rustup component add rustfmt
+  - if [ $TRAVIS_RUST_VERSION >= "1.36.0" ]; then
```
Comment (steveeJ) [inline .travis.yml]: I don't think the shell can compare semantic versions like this.

File: .travis.yml [hunk 2]
```diff
@@ -24,5 +24,11 @@ matrix:
...
+      cargo install --force cargo-audit;
+    fi
 script:
   - cargo fmt -- --check
+  - if [ $TRAVIS_RUST_VERSION >= "1.36.0" ]; then
```
Comment (steveeJ) [inline .travis.yml]: I don't think the shell can compare semantic versions like this.

**Comment by mbestavros** [comment]
@steveeJ Yes, you're right, my apologies. I've changed it so it ignores individual versions rather than trying to ignore any version below `1.36.0`; I don't see that being a problem unless we need a whole lot more exceptions, and I don't think we will.

**Comment by steveeJ** [comment]
> Yes, you're right, my apologies. I've changed it so it ignores individual versions rather than trying to ignore any version below `1.36.0`; I don't see that being a problem unless we need a whole lot more exceptions, and I don't think we will.

I meant that the `>=` operator does have version comparison semantics.

**Comment by mbestavros** [comment]
I forgot to push my changes... :stuck_out_tongue: 

Should be correct now.

**Comment by steveeJ** [comment]
I filed https://github.com/mbestavros/flagset/pull/1/files as a suggestion to correct the version detection. PTAL

**Comment by mbestavros** [comment]
@steveeJ I fixed a small syntax error and squashed down to 1 commit; tests pass now.

**Pull request merged**